src/components/scorecard/models/ScorecardOptInSchema.ts

```diff
@@ -0,0 +1,9 @@
+import { z } from "zod";
+
+export const scorecardOptInSchema = z.object({
+  firstName: z.string().nonempty(),
```

Comment:
the nonempty here should also be applied to email and it should appropriately show an error without your error clause in the `register` clause.

Reply:
The `.email()` method already protects against empty and whitespace strings. But, I added `.trim()` to `firstName`.